src/to_square_example.c

```diff
@@ -0,0 +1,177 @@
+#include <pebble.h>
+
+
+#define ABS(a) (((a) > 0) ? (a) : -1 * (a))
+#define IMAGES_LENGTH 2
+
+
+
+static Window *s_main_window;
+static Layer *icon_layer;
+static GDrawCommandImage* images [IMAGES_LENGTH];
+
+
+static int to_square_normalized;
+static int current_image;
+
+static int16_t prv_int_attract_to(int16_t i, int16_t bounds, int32_t normalized) {
+  const int16_t delta_0 = (int16_t) ((0 + 1) - i);
+  const int16_t delta_b = (int16_t) ((bounds - 1) - i);
+  const int16_t delta = ABS(delta_0) < ABS(delta_b) ? delta_0 : delta_b;
+
+  return (int16_t) (i + delta * normalized / ANIMATION_NORMALIZED_MAX);
+}
+
+GPoint gpoint_attract_to_square(GPoint point, GSize size, int32_t normalized) {
+  point.y += 1;
+  point = GPoint(
+      prv_int_attract_to(point.x, size.w, normalized),
+      prv_int_attract_to(point.y, size.h, normalized));
+  return point;
+}
+
+typedef struct {
+  GSize size;
+  int32_t normalized;
+} ToSquareCBContext;
+
+static bool prv_attract_draw_command_list_to_square_cb(GDrawCommand *command, uint32_t index, void *context) {
+
+  ToSquareCBContext *to_square = context;
+  for (int i = 0; i < gdraw_command_get_num_points(command); i++) {
+    gdraw_command_set_point(command, i, gpoint_attract_to_square(gdraw_command_get_point(command, i), to_square->size, to_square->normalized));
+  }
+  return true;
+}
+
+void attract_draw_command_list_to_square(GDrawCommandList *list, GSize size, int32_t normalized) {
+  ToSquareCBContext ctx = {
+      .size = size,
+      .normalized = normalized,
+  };
+  gdraw_command_list_iterate(list, prv_attract_draw_command_list_to_square_cb, &ctx);
+}
+
+void attract_draw_command_image_to_square(GDrawCommandImage *image, int32_t normalized) {
+  attract_draw_command_list_to_square(gdraw_command_image_get_command_list(image), gdraw_command_image_get_bounds_size(image), normalized);
+}
+
+//Update callback for the background layer
+
+static void bg_layer_update_proc(Layer *layer, GContext *ctx) {
+  GRect screen_rect = layer_get_bounds(layer);
+  graphics_context_set_fill_color(ctx, GColorRed);
+  graphics_fill_rect(ctx, screen_rect, 0, GCornerNone);
+}
+
+
+//Update callback for the icon layer
+
+static void icon_layer_update_proc(Layer *layer, GContext *ctx) {
+  if (!images[current_image]) {
+    return;
+  }
+
+  GDrawCommandImage *temp_copy = gdraw_command_image_clone(images[current_image]);
+  attract_draw_command_image_to_square(temp_copy, to_square_normalized);
+  graphics_context_set_antialiased(ctx, true);
+  gdraw_command_image_draw(ctx, temp_copy, GPoint(0, 0));
+  free(temp_copy);
```

Comment:
You should not directly free memory given to you by the API. Use `gdraw_command_image_destroy()` instead.